Libraries/LibGUI/ProcessChooser.cpp

```diff
@@ -91,10 +92,31 @@ ProcessChooser::ProcessChooser(const StringView& window_title, const StringView&
     };
 
     table_view.model()->update();
+
+    if (m_refresh_enabled) {
+        m_refresh_timer = add<Core::Timer>();
+
+        // Start the timer to update the processes
+        m_refresh_timer->start(m_refresh_interval);
+        m_refresh_timer->on_timeout = [&table_view] {
+            auto selection = table_view.selection();
+            table_view.model()->update();
+
+            // If there is already a selection, we select the same index
+            // FIXME We are selecting the same index instead of the PID
```

Comment:
Indexes are logically invalid after a model update and cannot be safely reused.

Reply:
Changed as suggested. Now the selection is based on PID